ui/apps/platform/src/services/CollectionsService.ts

```diff
@@ -188,12 +188,28 @@ export type CollectionDryRunResponse = {
  * @returns A list of deployments that are resolved by the collection.
  */
 export function dryRunCollection(
-    dryRunRequest: CollectionDryRunRequest
-    // TODO `ListDeployment` will make this impossible to paginate without loading the entire
-    // dataset client side. Ask [BE] if there is an efficient way to aggregate namespaces/clusters
-    // under a matching deployment name similar to the graphql query. An alternative might be to
-    // change the rendering of the list to not group deployments, but to sort alphabetically.
+    collectionRequest: CollectionRequest,
+    searchFilter: string,
+    page: number,
+    pageSize: number
 ): CancellableRequest<ListDeployment[]> {
+    const dryRunRequest: CollectionDryRunRequest = {
```

Comment:
What do you think of this change? I figured it would be nice to get the options object out of the component and into the service but I'm not sure if this call will eventually be used in other areas of the code and things like `sortOption` would need to be flexible

Reply:
Yeah I like this, breaking apart the `CollectionDryRunRequest` and making it so the caller doesn't have to construct it itself is a good change, and I think more inline with other service functions in the app.

Reply:
Ditto.

Wow, you never know when API payloads and responses will split versus join:
* Instead of split `{ collection, options }`
* It joins/merges `options` property with collection properties for `CollectionDryRunRequest`